### PR TITLE
Use getent instead of nslookup for starting scripts

### PR DESCRIPTION
**Pull request by jinfwhuang**
The nslookup only exit with status 0 with successful lookup of dns record for both forward and backward lookups. It would mean that the cluster's DNS setup has to support.

I would recommend that the lookup would only be a forward lookup. Something like getent host should work. I will submit a PR to fix that.

pravega#76

### CODE REVIEW

**Comment by amuraru** [comment]
Thanks for your PR @jinfwhuang
Interesting, are you saying your k8s dns provider is not supporting fw and backward lookups for the zk service?

**Comment by jinfwhuang** [comment]
I don't know the exact details. The evidence suggests so. @mtougeron could provide more details.

**Comment by amuraru** [comment]
Ok - let me test this in our env as well.
Also we should also remove dnsutils pkg installation from https://github.com/pravega/zookeeper-operator/blob/b5502fb55b8d71f837d2d18ed0ab0469e8a24059/docker/Dockerfile#L23 ?

File: pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go [hunk 1]
```diff
@@ -20,13 +20,13 @@ import (
...
-	DefaultZkContainerRepository = "pravega/zookeeper"
+	// Default images
+	//DefaultZkContainerRepository = "pravega/zookeeper"
+	//DefaultZkContainerVersion = "0.2.8"
 
-	// DefaultZkContainerVersion is the default tag used for for the zookeeper
-	// container
-	DefaultZkContainerVersion = "0.2.8"
+	// Image built by Adobe ad-cloud
+	DefaultZkContainerRepository = "cr.k8s.tubemogul.info/jinhuan/zookeeper"
```
Comment (amuraru) [inline pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go]: I suggest we keep the defaults here from
Pravega upstream and use your CR to override this if needed. This way - submitting a PR upstream would be easier

**Comment by mtougeron** [comment]
We had to block ANY & AAAA record lookups in coredns due to a limitation of some of our apps. (It isn't ideal but it's the world we live in) This causes the `nslookup` calls to fail.

```yaml
        template ANY AAAA . {
            rcode NXDOMAIN
        }
```

File: pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go (hunk 1 above)
Comment (jinfwhuang) [reply pkg/apis/zookeeper/v1beta1/zookeepercluster_types.go]: Sounds good. Updated it.

**Comment by jinfwhuang** [comment]
About dnsutils pkg, I believe `getent` is installed through dnsutils.

**Comment by amuraru** [comment]
Makes sense @mtougeron - I'll test in my env as well

**Comment by amuraru** [comment]
> About dnsutils pkg, I believe getent is installed through dnsutils.
 
i think getent is a builtin cmd in base ubuntu image and not provided by dnsutils. am i wrong?

**Comment by jinfwhuang** [comment]
`nc` is used in some of those scripts. dnsutils installation would be needed there.

**Review by amuraru — APPROVED** [review]

**Pull request merged**